std/functional.d

```diff
@@ -1562,3 +1565,45 @@ template forward(args...)
     int value = 3;
     auto x2 = bar(value); // case of OK
 }
+
+/**
+$(D auto apply(Tup)(Tup t)
+    if (isTuple!Tup && Parameters!callable.length == Tup.length))
+
+Applies the items of tuple `t` as arguments to the function `callable`.
+
+Returns:
+    The return value of the function `callable` for the arguments
+    represented by the items of `t`.
+
+*/
+template apply(alias callable)
+{
+    import std.typecons : isTuple;
+    auto apply(Tup)(Tup t)
+    if (isTuple!Tup && Parameters!callable.length == Tup.length)
```

Comment:
Instead of checking if the number of arguments matches (which might cause trouble with overload sets), why not check `__traits(compiles, callable(t.expand))`?